Savanna/Model/Animal.cs

```diff
@@ -0,0 +1,196 @@
+﻿namespace Savanna
+{
+    /// <summary>
+    /// Contains all common properties and behaviour of all types of animals.
+    /// </summary>
+    public abstract class Animal
+    {
+        /// <summary>
+        /// Animal ID.
+        /// </summary>
+        public int Id { get; set; }
+
+        /// <summary>
+        /// Coordinate X - row.
+        /// </summary>
+        public int XCoordinate { get; set; }
+
+        /// <summary>
+        /// Coordinate Y - column.
+        /// </summary>
+        public int YCoordinate { get; set; }
+
+        /// <summary>
+        /// Animal vision.
+        /// </summary>
+        public int Vision { get; set; }
+
+        /// <summary>
+        /// Indicates if animal is alive.
+        /// </summary>
+        public bool IsAlive { get; set; }
+
+        /// <summary>
+        /// Symbol for animal on a game field.
+        /// </summary>
+        public string? Letter { get; set; }
+
+        private static int id = 0;
+
+        /// <summary>
+        /// Generates new ID.
+        /// </summary>
+        /// <returns>Incremented ID.</returns>
+        protected static int GenerateId() => id++;
+
+        /// <summary>
+        /// Makes new animal, generates the new animal's coordinates and adds it to list.
+        /// </summary>
+        /// <param name="newAnimal">New animal.</param>
+        /// <param name="animals">List of animals.</param>
+        public static void AddAnimal(Animal newAnimal, List<Animal> animals)
+        {
+            Random random = new();
+
+            {
+                newAnimal.XCoordinate = random.Next(0, GameParameters.boardRows - 1);
+                newAnimal.YCoordinate = random.Next(0, GameParameters.boardColumns - 1);
+
+                if (!IsCellReserved(newAnimal.XCoordinate, newAnimal.YCoordinate, animals) && !Game.IsCellOnBoard(newAnimal.XCoordinate, newAnimal.YCoordinate))
+                {
+                    animals.Add(newAnimal);
+                }
+                else
+                {
+                    return;
+                }
```

Comment:
Code:
else
{
    return;
}

is obsolete.

Reply:
Code part removed.